protobuf_definitions/message_formats.proto

```diff
@@ -362,13 +375,144 @@ message PositionEstimate {
+message DepthSetPointCommand {
+  DepthSetPoint depth_set_point = 1; // depth set point to go to
+}
+
+// The TiltMainCameraCommand can set the desired camera tilt angle.
+message TiltMainCameraCommand {
+  TiltAngle tilt_angle = 1; // tilt angle of the camera (-30..30)
+}
+
+// The TiltServoCommand is used to set the tilt angle of the servo.
+message TiltServoCommand {
+  MultibeamServo tilt_angle = 1; // tilt angle for the servo, i.e. multibeam
+}
+
+// WaitForCommand is used to wait duringing a mission.
+message WaitForCommand {
+  float wait_for_seconds = 1; // wait for x seconds
+}
+
+// CameraCommands are used to control the camera from a mission.
+message CameraCommand {
+  CameraAction camera_action = 1; // camera command
+  float action_param = 2; // used for taking photos based on a time or distance interval
+}
+
+// GoToSurfaceCommand is used to go to the surface.
+message GoToSurfaceCommand {
+  optional float desired_speed = 1; // desired speed to surface (m/s)
+}
+
+// GoToSeaBottomCommand is used to go to the sea bottom.
+message GoToSeaBottomCommand {
+  optional float desired_speed = 1; // desired speed to seabed (m/s)
+}
+
+// GoToHomeCommand is used to go to the home position.
+message GoToHomeCommand {
+  optional float desired_speed = 1; // desired speed to home (m/s)
+}
+
+// List of available camera actions.
+enum CameraAction {
+  CAMERA_ACTION_UNSPECIFIED = 0; // Unspecified command
+  CAMERA_ACTION_TAKE_PHOTO = 1; // Take one photo
+  CAMERA_ACTION_TAKE_PHOTOS_TIME = 2; // Take a photo every x second
+  CAMERA_ACTION_TAKE_PHOTOS_DISTANCE = 3; // Take a photo every x meter
+  CAMERA_ACTION_STOP_TAKING_PHOTOS = 4; // Stop taking photos
+  CAMERA_ACTION_START_RECORDING = 5; // Start recording
+  CAMERA_ACTION_STOP_RECORDING = 6; // Stop recording
+}
+
+// Path segment used to describe segments of a mission as a line between to waypoints.
+message PathSegment {
+  uint32 id = 1; // path segment id starting at 0, -1 for inactive
+  float speed_to_target = 2; // desired speed over ground in (m/s)
+  float course_to_target = 3; // course to target relative to north (radians)
+  float depth_speed = 4; // desired speed in heave (m/s)
+  float horizontal_length = 5; // horizontal length of the path segment (m)
+  float vertical_length = 6; // vertical legth of the path segment (m)
+  uint32 from_wp_id = 7; // id of the starting waypoint
+  uint32 to_wp_id = 8; // id of the ending waypoint
+  float duration_time = 9; // estmated time it takes to complete given legth and desired speed (s)
+}
+
 // Heading source used during reset of the position estimate.
 enum HeadingSource {
   HEADING_SOURCE_UNSPECIFIED = 0; // Unspecified
   HEADING_SOURCE_DRONE_COMPASS = 1; // Uses the drone compass to set the heading
   HEADING_SOURCE_MANUAL_INPUT = 2; // Used when the user sets the heading manually
 }
 
+// Run Mission state
+message RunMissionState {
+  bool enabled = 1; // If the mission is running
```

Comment:
Does it make sense to have this as an enum?

Reply:
Yes I use it with false for pause. The alternative could be PauseMissionState of, but then we need more topics everywhere.

Reply:
I'll add PauseMissionCtrl and remove the bool.

We can just do the mapping in comm.